File: src/main/java/banner/model/Banner.java
package banner.model;

import org.springframework.format.annotation.NumberFormat;

import javax.validation.constraints.Max;

public class Banner {

    private Integer id;
    private String imgSrc;
    private Integer width;
    private Integer height;
    private String targetUrl;
    private Integer langId;
    private Integer priority;
    private Boolean activity;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getImgSrc() {
        return imgSrc;
    }

    public void setImgSrc(String imgSrc) {
        this.imgSrc = imgSrc;
    }

    public Integer getWidth() {
        return width;
    }

    public void setWidth(Integer width) {
        this.width = width;
    }

    public Integer getHeight() {
        return height;
    }

    public void setHeight(Integer height) {
        this.height = height;
    }

    public String getTargetUrl() {
        return targetUrl;
    }

    public void setTargetUrl(String targetUrl) {
        this.targetUrl = targetUrl;
    }

    public Integer getLangId() {
        return langId;
    }

    public void setLangId(Integer langId) {
        this.langId = langId;
    }

    public Integer getPriority() {
        return priority;
    }

    public void setPriority(Integer priority) {
        this.priority = priority;
    }

    public Boolean getActivity() {
        return activity;
    }

    public void setActivity(Boolean activity) {
        this.activity = activity;
    }

    @Override
    public String toString() {
        return "Banner{" +
                "id=" + id +
                ", imgSrc='" + imgSrc + '\'' +
                ", width=" + width +
                ", height=" + height +
                ", targetUrl='" + targetUrl + '\'' +
                ", langId=" + langId +
                ", priority=" + priority +
                ", activity=" + activity +
                '}';
    }
}

File: src/main/resources/static/js/components/banners/Banner.ts
import Model from "../Model";

export default class Banner extends Model{

    // public id : number | null = null;
    public imgSrc : string | null = null;
    public imgFile : any | null = null;
    public width : number | null = null;
    public height : number| null = null;
    public targetUrl : string | null = "http://";
    public langId : number | null = null;
    public priority : number | null = null;
    // public activity : boolean = true;

    public clean(){
        this.id = null;
        this.imgSrc = null;
        this.imgFile = null;
        this.width = null;
        this.height = null;
        this.targetUrl = "http://";
        this.langId = null;
        this.priority = null;
        this.activity = true;
    }

    copyItem<T extends Model>(localeRowFrom: T): void {
        this.id = localeRowFrom.id;
        this.imgSrc = (<Banner><unknown>localeRowFrom).imgSrc;
        this.width = (<Banner><unknown>localeRowFrom).width;
        this.height = (<Banner><unknown>localeRowFrom).height;
        this.targetUrl = (<Banner><unknown>localeRowFrom).targetUrl;
        this.langId = (<Banner><unknown>localeRowFrom).langId;
        this.priority = (<Banner><unknown>localeRowFrom).priority;
        this.activity = localeRowFrom.activity;
    }

    public static CLASS_NAME = 'Banner';
    public className : string = Banner.CLASS_NAME;

    // public static generalityOf<T extends Model>(item: T) : boolean {
    //     if (item.className === this.CLASS_NAME)
    //         return true;
    //     return super.generalityOf(item);
    // }
}